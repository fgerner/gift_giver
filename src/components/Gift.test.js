import React from "react";
import {shallow} from 'enzyme';
import toJson from "enzyme-to-json";
import Gift from "./Gift";

describe('Gift', () => {
    const gift = shallow(<Gift/>);

    it('renders properly', function () {
        expect(toJson(gift)).toMatchSnapshot();
    });

    it('initializes a person and present in `state`', () => {
        expect(gift.state()).toEqual({person:'', gift: ''})
    });

    describe('When typing into the person input', () => {
        const person = 'Wife';

        beforeEach(() => {
            gift.find('.input-person').simulate('change', {target: {value: person}});
        });
        it('should update the person in state', function () {
            expect(gift.state().person).toEqual(person);
        });
    });

    describe('when typing into the present input', () => {
        const present = 'Drone';
        
        beforeEach(() => {
            gift.find('.input-present').simulate('change', {target: {value: present}});
        })
        it('updates the present in `state`', () => {
            expect(gift.state().present).toEqual(present);
        })
    })
});


