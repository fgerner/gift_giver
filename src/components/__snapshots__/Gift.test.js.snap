// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift renders properly 1`] = `
<div>
  <Form
    inline={false}
  >
    <FormGroup>
      <Component>
        Person
      </Component>
      <FormControl
        className="input-person"
        onChange={[Function]}
      />
    </FormGroup>
  </Form>
</div>
`;
